### ISSUE TITLE
logistic regresstion

### ISSUE THREAD

**Issue by 16shery** [16shery/code-of-logistic-regression#1]
import numpy as np
from keras.datasets import mnist

# Load MNIST dataset
(X, y), (X_test, y_test) = mnist.load_data()

# Subset data to use only class 0 and class 1
indices = np.logical_or(y == 0, y == 1)
X = X[indices]
y = y[indices]

# Reshape images to 1D vectors
X = X.reshape(X.shape[0], -1)

# Standardize dataset
mean = np.mean(X)
std = np.std(X)
X_std = (X - mean) / std

# Shuffle data
np.random.seed(42)
shuffle_indices = np.random.permutation(X_std.shape[0])
X_std = X_std[shuffle_indices]
y = y[shuffle_indices]

# set the number of folds
num_folds = 10

# create the fold indices
fold_indices = np.array_split(np.arange(X_std.shape[0]), num_folds)

# define different learning rates to test
learning_rates = [0.001, 0.01, 0.1, 1]

# iterate over the learning rates and perform 10-fold cross-validation
for lr in learning_rates:
    accuracies = []
    for i in range(num_folds):
        val_indices = fold_indices[i]
        train_indices = np.concatenate(fold_indices[:i] + fold_indices[i+1:])

        X_train, X_val = X_std[train_indices], X_std[val_indices]
        y_train, y_val = y[train_indices], y[val_indices]

        # initialize weights to zero
        num_features = X_train.shape[1]
        w = np.zeros(num_features)

        # train logistic regression model with current learning rate
        num_epochs = 100
        for epoch in range(num_epochs):
            # calculate gradient
            z = np.dot(X_train, w)
            y_pred = 1 / (1 + np.exp(-z))
            error = y_pred - y_train
            gradient = np.dot(X_train.T, error)

            # update weights
            w -= lr * gradient

        # make predictions on validation set and calculate accuracy
        z_val = np.dot(X_val, w)
        y_pred_val = 1 / (1 + np.exp(-z_val))
        y_pred_val[y_pred_val < 0.5] = 0
        y_pred_val[y_pred_val >= 0.5] = 1
        accuracy = np.mean(y_pred_val == y_val)
        accuracies.append(accuracy)

    # report mean accuracy across all folds for the current learning rate
    mean_accuracy = np.mean(accuracies)
    print('Learning rate:', lr, 'Accuracy:', mean_accuracy)
